src/main/java/org/testng/internal/Utils.java

```diff
@@ -102,7 +102,11 @@ private Utils() {
 
   public static void writeUtf8File(String outputDir, String fileName, XMLStringBuffer xsb, String prefix) {
     try {
-      final File file = new File(outputDir, fileName);
+      final File outDir = (outputDir != null) ? new File(outputDir) : new File("").getAbsoluteFile();
```

Comment:
You change the previous logic, why not. But in this case, add `@Nullable` on `outputDir` like it is done on all other methods of the class.